CONTRIBUTORS.md

```diff
@@ -0,0 +1,118 @@
+## py4DSTEM Contributor Guidelines
+
+**Welcome to the py4DSTEM project!**
+
+We are grateful for your interest in contributing to py4DSTEM, an open-source Python library for 4D-STEM data analysis. Your contributions will help us make py4DSTEM a more powerful and versatile tool for the scientific community.
+
+This document outlines the guidelines and expectations for contributors to the py4DSTEM project. Please read it carefully before making any contributions.
+
+### Contribution Types
+
+There are many ways to contribute to py4DSTEM, including:
+
+* **Reporting bugs:** If you encounter a bug in py4DSTEM, please file a bug report on GitHub. Be sure to provide as much detail as possible about the bug, including steps to reproduce it.
+
+* **Submitting feature requests:** If you have a suggestion for a new feature for py4DSTEM, please submit a feature request on GitHub. Describe the feature in detail and explain how it would benefit users.
+
+* **Improving documentation:** py4DSTEM's documentation is always in need of improvement. If you have suggestions for improving the documentation, please submit a pull request or open an issue on GitHub.
+
+* **Developing new code:** If you are a developer, you can contribute to py4DSTEM by writing new code. Please follow the coding guidelines below.
+
+### Coding Guidelines
+
+* **Code style:** py4DSTEM uses the black code formatter and flake8 linter. All code must pass these checks without error before it can be merged. We suggest using `pre-commit` to help ensure any code commited follows these practices, checkout the [setting up developer environment section below](#install). We also try to abide by PEP8 coding style guide where possible.
+
+* **Documentation:** All code should be well-documented, and use Numpy style docstrings. Use docstrings to document functions and classes, add comments to explain complex code both blocks and individual lines; and use informative variable names.
+
+* **Testing:** Ideally all new code should be accompanied by tests using pyTest framework; but at the least we require examples of old and new behaviour caused by the PR. For bug fixes this can be a block of code which currently fails and works with the proposed changes. For new workflows or extensive feature additions,we'd prefer a Jupyter notebook demonstrating the changes for an entire workflow i.e. from loading the input data and saving any processed results. 
+
+* **Dependencies:** Adding dependencies should be done so thoughtfully. Prioritize adding dependencies that are actively maintained, have permissive installation requirements, and are accessible through both pip and conda.
+
+### Pull Requests
+
+When submitting a pull request, please:
+
+* **Open a corresponding issue:** Before submitting a pull request, open an issue to discuss your changes and get feedback.
+
```

Comment:
@sezelt so you're sugesting, altering this part? @gvarnavi

Would something like:
```
when submitting a pull request, please:

* **Bug Fixes and Errors - Open a correspinding issue :** If you're planning on submitting a PR to address a bug fix or error, open an issue to highlight the bug or error and get feedback.

* **New Feature or Large Changes - Open a draft PR:** If you're looking to add a new feature or make a substantive change to the code, open a draft PR to get feedback and discuss how to approach the changes.

```

Reply:
I think I would leave this section as-is.  I agree @gvarnavi that draft PRs are very useful for brainstorming new code / changes / features, and we can definitely use them this way.  That said, the primary purpose of this document is to get people who have not contributed before up and running, so I think just having one thing they do (set up an issue) might be better than giving options here, just to keep things simple to get folks going.  That wouldn't preclude using draft PRs.

Reply:
I'm good with leaving it as is or changing it; I think others may have stronger opinions on this. I'd say we should just pick one so we can close this PR. 
@sezelt  @gvarnavi - Do you have strong opinions, if so do you want to take a stab at editing it?

Reply:
I'm going to go ahead and accept this as-is for now :)